### PR TITLE
Collapses LogAccess and LogAccessHttp into LogAccess

### PR DESCRIPTION
**Pull request by randall**
Also, makes ProxyError::str inline so traffic_logstats/logcat doesn't need
to pull in a larger number of dependencies.

### CODE REVIEW

**Comment by zwoop** [comment]
[approve ci autest]

**Comment by zwoop** [comment]
Nice!

**Comment by d2r** [comment]
```
(docker:ats:centos:7)[~/trafficserver/lib/ts]
linuxkit-025000000001 ∴ make check
make  test_tsutil test_arena test_atomic test_freelist test_geometry test_List test_Map test_Vec test_X509HostnameValidator test_tslib
make[1]: Entering directory '/Users/derekd/devel/trafficserver/lib/ts'
make[1]: 'test_tsutil' is up to date.
make[1]: 'test_arena' is up to date.
make[1]: 'test_atomic' is up to date.
make[1]: 'test_freelist' is up to date.
make[1]: 'test_geometry' is up to date.
make[1]: 'test_List' is up to date.
make[1]: 'test_Map' is up to date.
make[1]: 'test_Vec' is up to date.
make[1]: 'test_X509HostnameValidator' is up to date.
make[1]: *** No rule to make target '../../iocore/eventsystem/libinkevent.a', needed by 'test_tslib'.  Stop.
make[1]: Leaving directory '/Users/derekd/devel/trafficserver/lib/ts'
make: *** [Makefile:2053: check-am] Error 2
```

I am seeing this at commit `ebac8be7b`. Do you see the same? Not sure if it is related.

**Comment by randall** [comment]
@d2r I can reproduce that error with a clean tree from tip of master as well. I think it's been that way for some time (at least 4 months).

**Comment by d2r** [comment]
> @d2r I can reproduce that error with a clean tree from tip of master as well. I think it's been that way for some time (at least 4 months).

Thanks. I think we need to run `make` before running `make check` so that the needed archive is available. Regardless it is not a problem for this pull request.

File: proxy/logging/LogAccess.cc [hunk 1]
```diff
@@ -20,1495 +20,2711 @@
...
 int
-LogAccess::marshal_http_header_field_escapify(LogField::Container /* container ATS_UNUSED */, char * /* field ATS_UNUSED */,
-                                              char *buf)
+LogAccess::marshal_client_finish_status_code(char *buf)
 {
-  DEFAULT_STR_FIELD;
+  if (buf) {
+    int code                                  = LOG_FINISH_FIN;
+    HttpTransact::AbortState_t cl_abort_state = m_http_sm->t_state.client_info.abort;
+    if (cl_abort_state == H
```
Comment (d2r) [inline proxy/logging/LogAccess.cc]: Note: I have confirmed that this code was "pulled up" from the old `LogAccessHttp` with only changes to the scope to remove the `Http`. The diff of LogAccess.cc here is large, but mostly it is moving unchanged code.

File: proxy/ProxyClientSession.h [hunk 2]
```diff
@@ -47,7 +47,23 @@ enum class ProxyErrorClass {
...
+      return 1;
+    }
+
+    buf[0] = (this->cls == ProxyErrorClass::SSN) ? 'S' : 'T';
+    ++len;
+
+    len += snprintf(buf + len, buf_len - len, "%" PRIx32, this->code);
+
+    return len;
+  }
```
[the remark this answers is not in the document]
Comment (randall) [reply proxy/ProxyClientSession.h]: Without moving this inline, the number of libs hav\ing to be pulled in would've been much larger (I didn't chase down to it's end though)

**Comment by randall** [comment]
And I cleaned up the reference to "LogAccessHttpSM"

**Comment by randall** [comment]
I would also suggest this gets merged into 8.0 (it currently applies cleanly). This would reduce anycomplications of back porting  logging related changes to this branch.

File: proxy/ProxyClientSession.h (hunk 2 above)
[the remark this answers is not in the document]
Comment (d2r) [reply proxy/ProxyClientSession.h]: OK fair enough. Thanks.

**Comment by zwoop** [comment]
Cherry-picked to 8.0.x

**Pull request merged**